torch/_dynamo/guards.py

```diff
@@ -496,13 +497,42 @@ def NN_MODULE(self, guard: Guard):
         else:
             self._produce_guard_code(guard, [f"{name}({ref})"])
 
+    def CODE_MATCH(self, guard: Guard):
+        """Checks that the `__code__` of two functions are equal"""
+        ref = self.arg_ref(guard)
+        # Cannot guard with local var deps
+        if re.match(r"^(.*)L\[.*\](.*)$", ref) or "L[" in ref:
+            return
+        value = self.get(guard.name)
+
+        if hasattr(value, "__code__") and isinstance(value.__code__, types.CodeType):
+            code = list()
+            code.append(f"hash({ref}.__code__.co_code) == {hash(value.__code__.co_code)}")
```

Comment:
Not sure if this will result in invalid export across different python versions that generate different bytecode?